### PR TITLE
Handling all Venia images with Widths

### PR DESCRIPTION
**Pull request by revanth0212**
## Description

Previously we were only handling catalog images with widths. The SW had a rule to only look for images with URL pathname that starts with `/media/catalog`. Firstly, that is configurable and on top of it, there can be other images that need this rule check as well. Hence I have changed the rule to look for `JPEG and PNG` images instead.

## Related Issue
Closes [PWA-944](https://jira.corp.magento.com/browse/PWA-944)

### Verification Stakeholders
@dpatil-magento 

### Verification Steps
1. Go to the deployed app.
2. Open dev tools and set the network status to offline.
3. Reload the page, the home page should render all images, including page builder content.

## Screenshots / Screen Captures (if appropriate)
image

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Review by jimbo — APPROVED** [review]
Makes sense to me. Curious if @zetlen has any objections or cautions about expanding this approach to all images.

**Comment by dpatil-magento** [comment]
@revanth0212 Please check if this needs to be fixed as well - Go to Homepage (don't scroll page) > Go to any category > Set offline mode > Hit Venia logo to go to Home page > Now scroll to bottom of page - 

image

**Review by larsroettig — APPROVED** [review]

**Comment by revanth0212** [comment]
> @revanth0212 Please check if this needs to be fixed as well - Go to Homepage (don't scroll page) > Go to any category > Set offline mode > Hit Venia logo to go to Home page > Now scroll to bottom of page -
> 
> image

Theoretically speaking, since the images are not in the viewport (if you dint scroll down while online), they won't be fetched initially. When you come back to the page in offline mode, the page tries to request it but the SW won't have them, and neither can it go to the server to fetch them. Hence it fails.

That said, in my case when I tried the same, some images loaded, and some dint. Is this browser specific? I am on chrome `86.0.4240.111`.

image
image

File: packages/venia-concept/src/ServiceWorker/registerRoutes.js [hunk 1]
```diff
@@ -17,34 +21,36 @@ import { THIRTY_DAYS, MAX_NUM_OF_IMAGES_TO_CACHE } from './defaults';
...
         );
     });
 
     /**
      * Route to handle all types of images. Stores them in cache with a
      * cache name "images". They auto expire after 30 days and only 60
      * can be stored at a time.
+     *
+     * There is another route that handles images without width and options on them.
+     * This route handles images that wont have width options on them.
```
Comment (zetlen) [inline packages/venia-concept/src/ServiceWorker/registerRoutes.js:48]: I'm a little confused by this comment. This route handles images that won't have width options on them, or that will?

**Comment by zetlen** [comment]
@jimbo I see no problem with it, as long as this is true:
- [ ] Pre-generated images that are processed by Webpack get pre-generated at reasonable sizes
- [ ] Non-photographic images that should be PNGs or GIFS are not accidentally turned into JPEGs
- [ ] No parameters go on images that aren't from the local domain or the CDN

As for the best practice of declaring widths on most or all images, I'd trust you, @jimbo over my own intuitions by now.

**Comment by revanth0212** [comment]
> @jimbo I see no problem with it, as long as this is true:
> 
> * [ ]  Pre-generated images that are processed by Webpack get pre-generated at reasonable sizes
> * [ ]  Non-photographic images that should be PNGs or GIFS are not accidentally turned into JPEGs
> * [ ]  No parameters go on images that aren't from the local domain or the CDN
> 
> As for the best practice of declaring widths on most or all images, I'd trust you, @jimbo over my own intuitions by now.

Hey JZ, thanks for the review. I can assure you that this PR doesn't change anything related to pre-generated images, image conversion, or adding new parameters.

**Comment by dpatil-magento** [comment]
QA Approved.

**Pull request merged**